### ISSUE TITLE
When building a sequential model, what is y_pred? The final output of the model?

### ISSUE THREAD

**Issue by zotroneneis** [keras-team/keras#8323]
When creating a model, like:

        model = Sequential()
        model.add(embedding_layer)
        model.add(LSTM(n_hidden, return_sequences=False))
        model.add(Dropout(dropout_keep_prob))
        model.add(Dense(vocab_size))
        model.add(Activation('softmax'))
        optimizer = RMSprop(lr=self.lr)

        model.compile(optimizer=optimizer, loss='sparse_categorical_crossentropy', metrics=['accuracy'])

The definition of the loss functions and metrics always has the form `loss_function(y_true, y_pred)` or `metric(y_true, y_pred)`.

What is `y_pred`? Is it always the output of the final model layer? So in the example above it would be softmax probabilities? When changing the final layer to model.add(Activation('linear')), does` y_pred `contain the unnormalized probabilties/ logits?

**Comment by AndrasEros** [comment]
The answers are yes to all your questions.
